core/hashfuncs.h

```diff
@@ -75,9 +76,31 @@ static inline uint32_t hash_djb2_one_float(float p_in,uint32_t p_prev=5381) {
 		uint32_t i;
 	} u;
 
-	// handle -0 case
-	if (p_in==0.0f) u.f=0.0f;
-	else u.f=p_in;
+	// Normalize +/- 0.0 and Nan values so they hash the same.
+	if (p_in==0.0f)
+		u.f=0.0;
+	else if (std::isnan(p_in))
+		u.f=std::numeric_limits<float>::quiet_NaN();
+	else
+		u.f=p_in;
+
+	return ((p_prev<<5)+p_prev)+u.i;
+}
+
+// Overload for real_t size changes
+static inline uint32_t hash_djb2_one_float(double p_in,uint32_t p_prev=5381) {
+	union {
+		double d;
+		uint64_t i;
+	} u;
+
+	// Normalize +/- 0.0 and Nan values so they hash the same.
+	if (p_in==0.0f)
+		u.d=0.0;
+	else if (std::isnan(p_in))
+		u.d=std::numeric_limits<double>::quiet_NaN();
+	else
+		u.d=p_in;
 
 	return ((p_prev<<5)+p_prev)+u.i;
```

Comment:
You are correct, I'll move the 64 bit hasher into hashfucs.h and reuse it for this case.